src/pages/RepoPage/FailedTestsTab/hooks/useInfiniteTestResults.tsx

```diff
@@ -0,0 +1,208 @@
+          edges {
+            node {
+              updatedAt
+              avgDuration
+              name
+              failureRate
+              commitsFailed
+            }
+          }
+          pageInfo {
+            endCursor
+            hasNextPage
+          }
+        }
+      }
+      ... on NotFoundError {
+        message
+      }
+      ... on OwnerNotActivatedError {
+        message
+      }
+    }
+  }
+}`
+
+interface UseTestResultsArgs {
+  provider: string
+  owner: string
+  repo: string
+  filters?: {
+    branch?: string
+  }
+  first?: number
+  after?: string
+  last?: number
+  before?: string
+  opts?: UseInfiniteQueryOptions<{
+    testResults: TestResult[]
+    pageInfo: { endCursor: string | null; hasNextPage: boolean }
+  }>
+}
+
+export const useInfiniteTestResults = ({
+  provider,
+  owner,
+  repo,
+  filters,
+  first = 20,
+  after,
+  last,
+  before,
+  opts = {},
+}: UseTestResultsArgs) => {
+  const { data, ...rest } = useInfiniteQuery({
+    queryKey: [
+      'GetTestResults',
+      provider,
+      owner,
+      repo,
+      filters,
+      first,
+      after,
+      last,
+      before,
+    ],
+    queryFn: ({ pageParam = after, signal }) =>
+      Api.graphql({
+        provider,
+        query,
+        signal,
+        variables: {
+          provider,
+          owner,
+          repo,
+          filters,
+          first,
+          after: pageParam,
+          last,
+          before,
+        },
+      }).then((res) => {
+        const parsedData = GetTestResultsSchema.safeParse(res?.data)
+
+        if (!parsedData.success) {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useInfiniteTestResults - 404 Failed to parse data',
+          } satisfies NetworkErrorObject)
+        }
+
+        const data = parsedData.data
+
+        if (data?.owner?.repository?.__typename === 'NotFoundError') {
+          return Promise.reject({
+            status: 404,
+            data: {},
+            dev: 'useInfiniteTestResults - 404 Not found error',
+          } satisfies NetworkErrorObject)
+        }
+
+        if (data?.owner?.repository?.__typename === 'OwnerNotActivatedError') {
+          return Promise.reject({
+            status: 403,
+            data: {
+              detail: (
+                <p>
+                  Activation is required to view this repo, please{' '}
+                  {/* @ts-expect-error */}
+                  <A to={{ pageName: 'membersTab' }}>click here </A> to activate
+                  your account.
+                </p>
+              ),
+            },
+            dev: 'useInfiniteTestResults - 403 Owner not activated',
+          } satisfies NetworkErrorObject)
+        }
+
+        return {
+          testResults:
+            data?.owner?.repository?.testResults?.edges?.map(
+              (edge) => edge.node
+            ) ?? [],
```

Comment:
`m`: We have a typesafe util function to do this mapping for you, and will handle all of the potential other cases for you as well: [`mapEdges`](https://github.com/codecov/gazebo/blob/main/src/shared/utils/graphql.ts#L12)

Reply:
i actually tried to use it but for some reason it fails when using it. I think something's off with the expected returned type in mapEdges i've seen this happen with useInfiniteQuery a couple of times.

Reply:
any tip is welcomed!

Reply:
You can do the following and it'll make everything happy:

```ts
testResults: mapEdges(data?.owner?.repository?.testResults).filter(
  (item): item is TestResult => item !== null
),
```